### PR TITLE
support puppet master setup on FreeBSD

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: manifests/server/rack.pp [hunk 1]
```diff
@@ -16,9 +16,13 @@
   $rack_arguments = $::puppet::server_rack_arguments,
   $user           = $::puppet::server_user,
   $vardir         = $::puppet::vardir,
+  $touch_cmd      = $::osfamily ? {
+    /^(FreeBSD|DragonFly)$/ => '/usr/bin/touch',
+    default                 => '/bin/touch'
```
Comment (mmoll) [inline manifests/server/rack.pp]: Not 100% sure about that, as I read that historically only RedHat/Fedora had this in /bin. However, Debian has a symlink by default, so this never got noticed.

File: manifests/server/install.pp [hunk 2]
```diff
@@ -2,9 +2,10 @@
 class puppet::server::install {
 
   $server_package_default = $::puppet::server_implementation ? {
-    'master'       => $::operatingsystem ? {
-      /(Debian|Ubuntu)/ => ['puppetmaster-common','puppetmaster'],
-      default           => ['puppet-server'],
+    'master'       => $::osfamily ? {
+      /(Debian|Ubuntu)/       => ['puppetmaster-common','puppetmaster'],
```
Comment (ekohl) [inline manifests/server/install.pp]: Debian should do if you're changing it to osfamily.

File: manifests/server/rack.pp (hunk 1 above)
Comment (ekohl) [inline manifests/server/rack.pp]: I'm not a fan of this complex logic in class parameters. Why not use ```touch``` as command and set a ```path``` which includes those directories?

Comment (mmoll) [inline manifests/server/rack.pp]: indeed, changed that. :bow:

File: manifests/server/install.pp (hunk 2 above)
Comment (mmoll) [inline manifests/server/install.pp]: it does ;)

**Comment by ekohl** [comment]
:+1:

File: .gitignore [hunk 3]
```diff
@@ -34,3 +34,6 @@ spec/fixtures/
 ## Puppet module
 pkg/
 coverage/
+
+# JetBrains IDEs
```
Comment (domcleal) [inline .gitignore]: You'll need to put this in modulesync, it's a synced file.

**Comment by domcleal** [comment]
:+1: otherwise.

File: .gitignore (hunk 3 above)
Comment (mmoll) [inline .gitignore]: fixed.

**Comment by ekohl** [comment]
Thanks!

**Pull request closed without merging**